tasks/main.yml

```diff
@@ -1,70 +1,41 @@
 ---
-# tasks file for users
-
 - name: "common | set os specific variables"
   include_vars: "{{ item }}"
   with_first_found:
    - "os-{{ ansible_distribution }}-{{ ansible_distribution_release }}.yml"
    - "os-{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
    - "os-{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
    - "os-{{ ansible_distribution }}.yml"
-   - "os-{{ ansible_os_family }}.yml"
    - "os-default.yml"
   tags: users
 
-- name: user group creation
-  action: group 
-    name="{{ item.username }}"
-    gid="{{ item.uid if item.uid is defined else users_available[item.username]['uid'] if users_available[item.username]['uid'] is defined else omit }}"
-  with_items: '{{ users }}'
-  when: users_create_user_group
-  tags: users
-
-- name: user creation
-  action: 
-    user
-    name={{ item.username }}
-    group={{ item.group if item.group is defined else users_available[item.username]['group'] if users_available[item.username]['group'] is defined else item.username }}
-    groups="{{ item.groups|join(',') if item.groups is defined else users_available[item.username]['groups']|join(',') if users_available[item.username]['groups'] is defined else omit }}"
-    shell="{{ item.shell|default(users_available[item.username]['shell'])|default("/bin/bash") }}"
-    comment="{{ item.comment if item.comment is defined else users_available[item.username]['comment'] if users_available[item.username]['comment'] is defined else users_available[item.username]['name'] }}"
-    uid="{{ item.uid|default(users_available[item.username]['uid']) if users_available[item.username]['uid'] is defined else omit }}"
-    createhome=yes
-    password="{{ item.password|default(users_available[item.username]['password'])|default(omit) }}"
-    update_password={{ users_update_password }}
-  with_items: '{{ users }}'
-  tags: users
-
-- name: is user admin
-  action:
-    user
-    name={{ item.username }}
-    groups="{{ users_admin_groups|join(',') }}"
-    append=yes
-  when: "item.is_admin|default(users_available[item.username]['is_admin'])|default(False)"
+- name: create
+  include:
+    user_create.yml user={{ user_item }}
   with_items: '{{ users }}'
-  tags: users
-
-
-- name: creating .ssh directory
-  action: file path=~{{ item.username }}/.ssh owner={{ item.username }} group={{ item.group if item.group is defined else users_available[item.username]['group'] if users_available[item.username]['group'] is defined else item.username }} state=directory mode=0750
+  loop_control:
```

Comment:
It's not what I meant)
You should have left loop_control for `user_sshkey.yml` because that construction was necessary (made sense) for `user_sshkey.yml` 'cause there was inner loop.